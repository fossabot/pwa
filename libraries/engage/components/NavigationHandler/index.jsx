import React from 'react';
import PropTypes from 'prop-types';
import { UIEvents } from '@shopgate/pwa-core';
import {
  NAVIGATION_PUSH,
  NAVIGATION_POP,
  NAVIGATION_REPLACE,
  NAVIGATION_RESET,
} from '../../core/router/helpers';
import connect from './connector';

/**
 * Handles
 */
class NavigationHandler extends React.PureComponent {
  static propTypes = {
    children: PropTypes.node.isRequired,
    pop: PropTypes.func,
    push: PropTypes.func,
    replace: PropTypes.func,
    reset: PropTypes.func,
  };

  static defaultProps = {
    pop: () => { },
    push: () => { },
    replace: () => { },
    reset: () => { },
  }

  /**
   * @param {Object} props The component props.
   */
  constructor(props) {
    super(props);

    UIEvents.addListener(NAVIGATION_PUSH, props.push);
    UIEvents.addListener(NAVIGATION_POP, props.pop);
    UIEvents.addListener(NAVIGATION_REPLACE, props.replace);
    UIEvents.addListener(NAVIGATION_RESET, props.reset);
  }

  /**
   * Removes the event listeners.
   */
  componentWillUnmount() {
    UIEvents.removeListener(NAVIGATION_PUSH, this.props.push);
    UIEvents.removeListener(NAVIGATION_POP, this.props.pop);
    UIEvents.removeListener(NAVIGATION_REPLACE, this.props.replace);
    UIEvents.removeListener(NAVIGATION_RESET, this.props.reset);
  }

  /**
   * @returns {null}
   */
  render() {
    return this.props.children;
  }
}

export default connect(NavigationHandler);
