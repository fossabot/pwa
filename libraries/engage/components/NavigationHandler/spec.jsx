import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { createMockStore } from '@shopgate/pwa-common/store';
import NavigationHandler from './index';

describe('<NavigationHandler />', () => {
  const store = createMockStore();
  it('should render as expected', () => {
    const wrapper = shallow((
      <Provider store={store}>
        <NavigationHandler>
          <div>Some content</div>
        </NavigationHandler>
      </Provider>
    ));
    expect(wrapper).toMatchSnapshot();
  });
});
