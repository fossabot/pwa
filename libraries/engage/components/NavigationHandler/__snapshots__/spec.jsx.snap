// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationHandler /> should render as expected 1`] = `
<Connect(NavigationHandler)>
  <div>
    Some content
  </div>
</Connect(NavigationHandler)>
`;
